src/orchestrator/orchestrator.rs

```diff
@@ -0,0 +1,281 @@
+use futures::future::join_all;
+use std::{cmp::min, error::Error};
+use tokio::sync::mpsc;
+use tokio_util::sync::CancellationToken;
+
+use super::entry_csv_parser::entry_parser;
+use crate::aggregation_merkle_sum_tree::AggregationMerkleSumTree;
+use crate::executor::ExecutorSpawner;
+
+pub struct Orchestrator<const N_ASSETS: usize, const N_BYTES: usize> {
+    executor_spawner: Box<dyn ExecutorSpawner>,
+    entry_csvs: Vec<String>,
```

Comment:
I found a method to read a single CSV file using multiple threads, but it's a bit tricky.
 For now, I've left it to use multiple CSV files, even though it's not as convenient for the user. I plan to create an issue to address this.

Reply:
> Isn't it possible to just pass a single CSV file and then literally split the file in chunks?

Yes, it is indeed possible to pass a single large CSV file and split it into smaller chunks.
However, this approach requires additional storage space and can be more time-consuming.

For instance, when splitting a large CSV file containing 2^28 entries into chunks of 2^14 entries using just the `awk` command, it took around 87 seconds to complete the process. I believe a similar result might be expected from a Rust code if it doesn't employ multi-threading.

I assumed that to reduce this processing time, exchanges have the ability to download multiple smaller CSV files directly from their databases instead of working with a single large file.